File: pset6/readability.py
def main():
    # Get input text
    text = input("Text: ")

    # Count number of letters, words and sentences
    letters = 0
    words = 1
    sentences = 0

    for c in text:
        # print(c)
        if c.isalpha():
            letters += 1
        elif c in '.?!':
            sentences += 1
        elif c == ' ':
            words += 1

    # Get Coleman-Liau index of the input text
    grade = coleman_liau(letters, words, sentences)

    if grade < 1:
        print("Before Grade 1")
    elif grade < 16:
        print(f"Grade {grade}")
    else:
        print("Grade 16+")


def coleman_liau(letters, words, sentences):
    """
    Function to calculate the Coleman-Liau index of a text.
    """
    L = letters / words * 100
    S = sentences / words * 100

    return round((0.0588 * L) - (0.296 * S) - 15.8)


main()

File: pset2/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>


int coleman_liau(int letters, int words, int sentences);

int main(void)
{
    int letters = 0;
    int words = 1;
    int sentences = 0;
    int grade;

    // Get user input
    string text = get_string("Text: ");

    // Count the number of letters, words and sentences in the text
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            letters++;
        }
        else if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            sentences++;

        }
        else if (text[i] == ' ')
        {
            words++;
        }
    }

    // Calculate grade based on number of letters, words and sentences
    grade = coleman_liau(letters, words, sentences);

    // Determine the output to the screen based on the grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        printf("Grade 16+\n");
    }

}

int coleman_liau(int letters, int words, int sentences)
{
    // Calculate the Coleman-Liau index for an input text,
    // based on the number of letters, words and sentences in the text.
    float L;
    float S;

    L = ((float) letters / (float) words) * 100;
    S = ((float) sentences / (float) words) * 100;

    return round((0.0588 * L) - (0.296 * S) - 15.8);
}
